xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -563,9 +597,12 @@ where
                                 "Error validating commit for own message. Intent ID [{}]: {err:?}",
                                 intent.id,
                             );
-                            // Return before merging commit since it does not pass validation
-                            // Return OK so that the group intent update is still written to the DB
-                            return Err(Ok(IntentState::Error));
+                            return Err(IntentValidationError {
+                                processing_error: GroupMessageProcessingError::CommitValidation(
+                                    err,
+                                ),
+                                next_intent_state: IntentState::Error,
+                            });
```

Comment:
This is another functional change - `CommitValidation` could return a retriable error (as it makes DB calls that could lock). Before, the intent would be marked as errored and we wouldn't apply the commit, which could cause a fork. Now, the caller will see that it is a retriable error and bubble it up so that it can be retried.

Reply:
I'm wrong that there are no retriable errors returned - `CommitValidationError` might return a retriable error above. In that case the change here might actually help prevent forks.